src/XR/features/WebXRControllerMovement.ts

```diff
@@ -0,0 +1,547 @@
+            registeredComponents: RegisteredComponent[];
+        };
+    } = {};
+
+    private _currentRegistrationConfigurations: WebXRControllerMovementRegistrationConfiguration[] = [];
+    // Feature configuration is syncronized - this is passed to all handlers (reduce GC pressure).
+    private _featureContext: WebXRControllerMovementFeatureContext;
+    // forward direction for movement, which may differ from viewer pose.
+    private _movementDirection: Nullable<Quaternion> = null;
+    private _movementState: WebXRControllerMovementState;
+    private _xrInput: WebXRInput;
+
+    // unused
+    private _rotationMatrix: Matrix = Matrix.Identity();
+    private _translationDirection: Vector3 = new Vector3();
+    private _movementTranslation: Vector3 = new Vector3();
+
+    /**
+     * The module's name
+     */
+    public static readonly Name = WebXRFeatureName.MOVEMENT;
+
+    /**
+     * Standard controller configurations.
+     */
+    public static readonly REGISTRATIONS: Record<string, WebXRControllerMovementRegistrationConfiguration[]> = {
+        "default": [
+            {
+                allowedComponentTypes: [WebXRControllerComponent.THUMBSTICK_TYPE, WebXRControllerComponent.TOUCHPAD_TYPE],
+                handedness: "left",
+                axisChangedHandler: (axes: IWebXRMotionControllerAxesValue, movementState: WebXRControllerMovementState, featureContext: WebXRControllerMovementFeatureContext, xrInput: WebXRInput) => {
+                    movementState.rotateX = Math.abs(axes.x) > featureContext.rotationThreshold
+                        ? axes.x
+                        : 0;
+                    movementState.rotateY = Math.abs(axes.y) > featureContext.rotationThreshold
+                        ? axes.y
+                        : 0;
+                }
+            }, {
+                allowedComponentTypes: [WebXRControllerComponent.THUMBSTICK_TYPE, WebXRControllerComponent.TOUCHPAD_TYPE],
+                handedness: "right",
+                axisChangedHandler: (axes: IWebXRMotionControllerAxesValue, movementState: WebXRControllerMovementState, featureContext: WebXRControllerMovementFeatureContext, xrInput: WebXRInput) => {
+                    movementState.moveX = (Math.abs(axes.x) > featureContext.movementThreshold)
+                        ? axes.x
+                        : 0;
+                    movementState.moveY = (Math.abs(axes.y) > featureContext.movementThreshold)
+                        ? axes.y
+                        : 0;
+                }
+            }
+        ]
+    };
+
+    /**
+     * The (Babylon) version of this module.
+     * This is an integer representing the implementation version.
+     * This number does not correspond to the webxr specs version
+     */
+    public static readonly Version = 1;
+
+    /**
+     * An event triggered after XR frame completes
+     */
+    public onAfterXRFrameObservable = new Observable<WebXRControllerMovementState>();

```

Comment:
How is this different than adding a new onFrame observable after this one was created?
What will be the use for those callbacks?

Also, I think naming here is not entirely correct - it's after this component was done, not after (or before) the xr frame

Reply:
removed those observables